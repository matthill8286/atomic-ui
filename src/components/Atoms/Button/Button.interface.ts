import * as H from 'history'
import React from 'react'

export type ButtonActionType =
  | 'ghost'
  | 'primary'
  | 'prominent'
  | 'secondary'
  | 'inverted'
  | 'outlined'
  | 'lightBorder'
  | 'darkBorder'

export type ButtonSize = 'lg' | 'md' | 'sm'

export type ButtonType = 'button' | 'reset' | 'submit'

export interface StyledButtonProps {
  actionType: ButtonActionType
  disabled: boolean
  isLoading: boolean
  isFlat?: boolean
  size: ButtonSize
  fullWidth?: boolean
  round?: boolean
  squared?: boolean
  curved?: boolean
  elevated?: boolean
  weight?: string
}

export interface ButtonProps
  extends React.AnchorHTMLAttributes<HTMLAnchorElement | HTMLButtonElement> {
  actionType?: ButtonActionType
  children?: React.ReactNode
  disabled?: boolean
  fullWidth?: boolean
  classNameWrapper?: string
  isLoading?: boolean
  isFlat?: boolean
  onClick?: (ev: React.MouseEvent<HTMLElement>) => void
  size?: ButtonSize
  to?: H.LocationDescriptor
  type?: ButtonType
  round?: boolean
  squared?: boolean
  id?: string
  curved?: boolean
  elevated?: boolean
  weight?: string
  showCheckmark?: boolean
}
