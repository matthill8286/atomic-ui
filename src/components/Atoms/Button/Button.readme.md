## Button atom

Use buttons to trigger actions and links. Buttons can contain a combination of a clear label and an icon while links are always text.
