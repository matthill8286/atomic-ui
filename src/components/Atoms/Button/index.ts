export * from './Button'
export * from './Button.interface'
export * from './IconButton'
