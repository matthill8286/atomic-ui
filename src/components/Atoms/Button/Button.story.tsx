import { withState } from '@dump247/storybook-state'
import { boolean, radios, text } from '@storybook/addon-knobs'
import { storiesOf } from '@storybook/react'
import * as React from 'react'
import { styled } from '@/styles'
import {
  StyleguideArticle,
  StyleguideBookmarkActive,
  StyleguideBookmarkOutlined,
} from '@excelwithbusiness/webmobile-svg-library'
import { Icon } from '../Icon'
import { Status } from '../Status'
import { Typo } from '../Typography'
import { Button } from './Button'
import { ButtonActionType, ButtonProps, ButtonSize } from './Button.interface'
import Readme from './Button.readme.md'

const StyledButtonGroupContainer = styled('div')`
  flex: 1;
  display: flex;
  flex-direction: row;
`

const StyledButtonGroup = styled('div')`
  flex: 1;
`

const actionTypes: { [key: string]: ButtonActionType } = {
  ghost: 'ghost',
  primary: 'primary',
  prominent: 'prominent',
  secondary: 'secondary',
  inverted: 'inverted',
  outlined: 'outlined',
  lightBorder: 'lightBorder',
  darkBorder: 'darkBorder',
}

const sizes: { [key: string]: ButtonSize } = { lg: 'lg', md: 'md', sm: 'sm' }

const knobs = (): ButtonProps => {
  return {
    actionType: radios('Type', actionTypes, actionTypes.prominent),
    children: text('Content', 'Button'),
    size: radios('Size', sizes, sizes.md),
    isLoading: boolean('Loading', false),
    isFlat: boolean('Flattened', false),
    elevated: boolean('Elevated', false),
    weight: text('Font weight', 'bold'),
  }
}

const knobsWithCheckmark = () => {
  return {
    ...knobs(),
    showCheckmark: boolean('Loading Complete', false),
  }
}

storiesOf('Design System/Atoms/Button', module)
  .add(
    'Default',
    () => {
      return <Button {...knobsWithCheckmark()} />
    },
    {
      info: Readme,
    }
  )
  .add(
    'Disabled',
    () => {
      return <Button {...knobs()} disabled />
    },
    {
      info: Readme,
    }
  )
  .add(
    'Full width',
    () => {
      return <Button {...knobsWithCheckmark()} fullWidth />
    },
    {
      info: Readme,
    }
  )
  .add(
    'Anchor',
    () => {
      return <Button href="#" {...knobs()} />
    },
    {
      info: Readme,
    }
  )
  .add(
    'Router link',
    () => {
      return <Button to="#" {...knobs()} />
    },
    {
      info: Readme,
    }
  )
  .add(
    'Round',
    () => {
      return (
        <Button round size={'lg'} {...knobs()}>
          NO
        </Button>
      )
    },
    {
      info: Readme,
    }
  )
  .add(
    'Curved',
    () => {
      return (
        <Button curved size={'lg'} {...knobsWithCheckmark()}>
          Button text
        </Button>
      )
    },
    {
      info: Readme,
    }
  )
  .add(
    'Squared',
    () => {
      return (
        <Button squared size={'lg'} {...knobs()}>
          YES
        </Button>
      )
    },
    {
      info: Readme,
    }
  )
  .add('with Icon', () => {
    return (
      <>
        <Button {...knobs()}>
          <Icon width={16} height={16}>
            <StyleguideBookmarkActive />
          </Icon>
          Icon on left side
        </Button>
        &nbsp;
        <Button {...knobs()}>
          Icon on right side
          <Icon color="white" width={24} height={24}>
            <StyleguideArticle fill="white" />
          </Icon>
        </Button>
        &nbsp;
        <Button {...knobs()} squared>
          <Icon width={16} height={16}>
            <StyleguideBookmarkActive />
          </Icon>
        </Button>
        &nbsp;
        <Button {...knobs()} round>
          <Icon width={16} height={16}>
            <StyleguideBookmarkOutlined />
          </Icon>
        </Button>
      </>
    )
  })
  .add(
    'Bookmark ',
    withState({ isAddedToBookmark: false }, store => {
      return store.state.isAddedToBookmark ? (
        <Button to="#" actionType={'primary'}>
          <Icon width={16} height={16}>
            <StyleguideBookmarkActive />
          </Icon>
          Added to Bookmark
        </Button>
      ) : (
        <Button
          actionType={'secondary'}
          onClick={() => store.set({ isAddedToBookmark: !store.state.isAddedToBookmark })}>
          <Icon width={16} height={16}>
            <StyleguideBookmarkActive />
          </Icon>
          Add to Bookmark
        </Button>
      )
    })
  )
  .add('Bookmark Full Width', () => {
    return (
      <Button actionType={'primary'} size="lg" fullWidth>
        <Icon width={16} height={16}>
          <StyleguideBookmarkActive />
        </Icon>
        Bookmark
      </Button>
    )
  })
  .add('with Status', () => {
    return (
      <>
        <Button round>
          <Icon width={16} height={16}>
            <StyleguideBookmarkOutlined />
          </Icon>
          <Status type="black">1</Status>
        </Button>
        &nbsp;
        <Button>
          Checkmark Status
          <Icon width={16} height={16}>
            <StyleguideArticle />
          </Icon>
          <Status type="white">1</Status>
        </Button>
      </>
    )
  })
  .add('with 3 Button Group', () => {
    return (
      <StyledButtonGroupContainer>
        <StyledButtonGroup>
          <Button fullWidth actionType="ghost">
            <Typo fontSize="sm" tag="span" color="grey5">
              Test
            </Typo>
          </Button>
        </StyledButtonGroup>
        <StyledButtonGroup>
          <Button fullWidth actionType="ghost">
            Test
          </Button>
        </StyledButtonGroup>
        <StyledButtonGroup>
          <Button fullWidth actionType="ghost">
            Test
          </Button>
        </StyledButtonGroup>
      </StyledButtonGroupContainer>
    )
  })
