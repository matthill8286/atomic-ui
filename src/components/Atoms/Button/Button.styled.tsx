import { darken } from 'polished'
import React, { FunctionComponent } from 'react'
import { Link as RouterLink } from 'react-router-dom'
import { media } from '@/styles/media'
import { css, styled } from '@/styles/styled'
import { Icon } from '../Icon'
import { LoadingIndicator } from '../LoadingIndicator'
import { Status } from '../Status'
import { ButtonProps, StyledButtonProps } from './Button.interface'

const buttonStyles = css<StyledButtonProps>`
  ${Icon} {
    padding: 0
      ${({ theme, actionType }) => (actionType !== 'outlined' ? theme.defaultSpacing : 0)}px;
  }

  ${Status} {
    position: absolute;
    top: -6px;
    right: -6px;
  }

  display: flex;
  justify-content: center;
  align-items: center;
  border-radius: ${({ round, theme }) => (round ? `100%` : theme.dimension.borderRadius1)};
  box-sizing: border-box;
  outline: none;
  cursor: ${({ disabled, isFlat }) => (disabled ? 'not-allowed' : isFlat ? 'default' : 'pointer')};
  appearance: none;
  -webkit-appearance: none;
  padding: ${({ round, squared }) => (round || squared ? '0' : '12px 24px')};
  position: relative;
  text-decoration: none;
  font-family: ${({ theme }) => theme.font.family.default};
  font-weight: ${({ weight }) => weight || 'normal'};
  font-size: ${({ theme }) => theme.font.size.sm};
  height: ${({ size, theme }) => theme.button.size[size]};
  width: fit-content;

  box-shadow: ${({ elevated, theme }) => (elevated ? theme.dimension.elevationLevel2 : 'none')};

  ${({ size, round, theme, squared }) =>
    (round || squared) &&
    `
    width: ${theme.button.size.sm};
    height: ${theme.button.size.sm};

    ${media.md} {
      width: ${theme.button.size[size]};
      height: ${theme.button.size[size]};
    }
  `}

  ${({ disabled, theme, isFlat }) =>
    !disabled &&
    !isFlat &&
    `
    :hover {
      box-shadow: ${theme.dimension.elevationLevel3};
    }`}

  ${({ fullWidth }) =>
    fullWidth &&
    `
    width: 100%;
  `}
  
  ${({ theme, curved }) => curved && `border-radius: ${theme.dimension.borderRadius7};`}

  ${({ actionType, disabled, isLoading, theme, isFlat }) =>
    (actionType === 'prominent' &&
      (disabled
        ? `
        background: ${theme.color.grey4};
        border: 2px solid ${theme.color.grey4};
        color: ${theme.color.white};
        opacity: ${theme.polished.inactive};`
        : `
        background: ${theme.color.primary};
        border: 2px solid ${theme.color.primary};
        color: ${isLoading ? 'transparent' : theme.color.white};

        :hover {
          background: ${!isFlat && darken(theme.polished.darken, theme.color.primary)};
          border: 2px solid ${!isFlat && darken(theme.polished.darken, theme.color.primary)};
        }`)) ||
    (actionType === 'primary' &&
      (disabled
        ? `
          background: ${theme.color.primary};
          border: 2px solid ${theme.color.primary};
          color: ${theme.color.grey2};
          opacity: ${theme.polished.inactive};`
        : `
          background: ${theme.color.primary};
          border: 2px solid ${theme.color.primary};
          color: ${isLoading ? 'transparent' : theme.color.white}`)) ||
    (actionType === 'secondary' &&
      `background: ${theme.color.white};

      ${
        disabled
          ? `
          border: 2px solid ${theme.color.grey4};
          color: ${theme.color.white};
          opacity: ${theme.polished.inactive};`
          : `
          border: 2px solid ${theme.color.grey5};
          color: ${isLoading ? 'transparent' : theme.color.grey5};`
      }`) ||
    (actionType === 'ghost' &&
      `border: 2px solid ${theme.color.white};
       background: ${theme.color.white};

      ${
        disabled
          ? `
        color: ${theme.color.grey5};
        opacity: ${theme.polished.inactive};`
          : `
        color: ${isLoading ? 'transparent' : theme.color.grey5};

        :hover {
          box-shadow: ${!isFlat && theme.dimension.elevationLevel3};
        }`
      }
    `) ||
    (actionType === 'inverted' &&
      `border: 2px solid ${theme.color.white};
       background: ${theme.color.grey4};
       
    ${
      disabled
        ? `
      color: ${theme.color.white};
      opacity: ${theme.polished.inactive};`
        : `
      color: ${isLoading ? 'transparent' : theme.color.white};

      :hover {
        box-shadow: none;
        background: ${theme.color.grey4};
      }`
    }
  `) ||
    (actionType === 'outlined' &&
      `border: 2px solid ${theme.color.primary};
       background: ${theme.color.white};

    ${
      disabled
        ? `
      opacity: ${theme.polished.inactive};
      :hover {
        box-shadow: none;
      }
        `
        : `
        color: ${isLoading ? 'transparent' : theme.color.primary};
      `
    }
  `) ||
    (actionType === 'lightBorder' &&
      `border: 1px solid ${theme.color.grey2};
       background: ${theme.color.white};
       border-radius: ${theme.dimension.borderRadius7};
    ${
      disabled
        ? `
      opacity: ${theme.polished.inactive};
      :hover {
        box-shadow: none;
      }
        `
        : `
      color: ${isLoading ? 'transparent' : theme.color.grey5};
      `
    }
  `) ||
    (actionType === 'darkBorder' &&
      `border: 2px solid ${theme.color.grey4};
       background: ${theme.color.white};

    ${
      disabled
        ? `
      opacity: ${theme.polished.inactive};
      :hover {
        box-shadow: none;
      }
        `
        : `
        color: ${isLoading ? 'transparent' : theme.color.grey5};
      `
    }
  `)}
`

export const StyledAnchorButton = styled.a<StyledButtonProps>`
  ${buttonStyles};

  -webkit-appearance: none !important;

  :visited,
  :hover,
  :active {
    box-shadow: none;
  }
`

export const StyledButton = styled.button<StyledButtonProps>`
  ${buttonStyles};

  > span {
    color: inherit;
    display: block;
    text-align: center;
    width: 100%;
  }

  ${({ actionType, disabled, isLoading, theme }) =>
    actionType === 'primary' &&
    (disabled
      ? `color: ${theme.color.grey2};`
      : `color: ${isLoading ? 'transparent' : theme.color.white}`)}
`

const RouterLinkWrapper: FunctionComponent<ButtonProps> = ({ children, className, to = '#' }) => (
  <RouterLink className={className} to={to}>
    {children}
  </RouterLink>
)

export const StyledRouterButton = styled(RouterLinkWrapper)`
  ${buttonStyles};
`

export const StyledLoadingIndicator = styled(LoadingIndicator)`
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  margin: auto;
`
