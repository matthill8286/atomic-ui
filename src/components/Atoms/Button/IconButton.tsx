import * as React from 'react'
import { styled } from '@/styles/styled'
import { Button } from './Button'
import { ButtonProps } from './Button.interface'

export interface IconButtonProps extends ButtonProps {
  width?: number | string
  height?: number | string
  isInputIcon?: boolean
  isFlat?: boolean
}

export interface StyledIconButtonProps extends IconButtonProps {
  width: number | string
  height: number | string
  isFlat?: boolean
}

export const StyledIconButton = styled(Button)<StyledIconButtonProps>`
  min-width: ${({ width, theme }) =>
    typeof width === 'number' ? `${width}px` : theme.spacing.base[width] || width};
  height: ${({ height, theme }) =>
    typeof height === 'number' ? `${height}px` : theme.spacing.base[height] || height};
  padding: 0 ${({ theme }) => theme.spacing.base.xs};
`

export const IconButton: React.FC<IconButtonProps> = ({
  actionType = 'outlined',
  height = 40,
  width,
  color,
  isLoading,
  isFlat,
  onClick,
  ...rest
}) => {
  return (
    <StyledIconButton
      color={color}
      actionType={actionType}
      width={width || height}
      height={height}
      isFlat={isFlat}
      isLoading={isLoading}
      onClick={onClick}
      {...rest}
    />
  )
}

IconButton.displayName = 'IconButton'
