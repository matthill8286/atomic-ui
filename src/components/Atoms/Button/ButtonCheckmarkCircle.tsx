import React from 'react'
import { IconCheckmarkCircle } from '@matthill8286/jsx-icon-library'
import { Icon } from '@/components/Atoms/Icon'

export const ButtonCheckmarkCircle: React.FC = () => (
  <Icon width={16} height={16}>
    <IconCheckmarkCircle />
  </Icon>
)
