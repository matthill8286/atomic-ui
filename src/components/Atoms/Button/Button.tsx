import * as React from 'react'
import { LoadingIndicatorColor } from '../LoadingIndicator'
import { ButtonActionType, ButtonProps } from './Button.interface'
import {
  StyledAnchorButton,
  StyledButton,
  StyledLoadingIndicator,
  StyledRouterButton,
} from './Button.styled'
import { ButtonCheckmarkCircle } from './ButtonCheckmarkCircle'

const buttonColours = {
  secondary: 'black',
  inverted: 'black',
  lightBorder: 'black',
  darkBorder: 'black',
  ghost: 'black',
  outlined: 'primary',
  primary: 'white',
  prominent: 'white',
}

export const Button: React.FC<ButtonProps> = ({
  actionType = 'prominent',
  children,
  disabled = false,
  round = false,
  squared = false,
  curved = false,
  fullWidth = false,
  isLoading = false,
  isFlat = false,
  elevated = false,
  showCheckmark = false,
  weight,
  onClick,
  size = 'md',
  to,
  href,
  type = 'button',
  className,
  id,
  ...otherProps
}) => {
  const handleOnClick = (ev: React.MouseEvent<HTMLElement>) => {
    ev.stopPropagation()
    if (disabled || isLoading) {
      return
    }
    if (onClick) {
      onClick(ev)
    }
  }

  const renderLoading = (actionType: ButtonActionType) => {
    const color = buttonColours[actionType] as LoadingIndicatorColor
    return <StyledLoadingIndicator isVisible={true} size={20} color={color} />
  }

  if (href) {
    return (
      <StyledAnchorButton
        actionType={actionType}
        size={size}
        type={type}
        fullWidth={fullWidth}
        disabled={disabled}
        round={round}
        squared={squared}
        curved={curved}
        elevated={elevated}
        weight={weight}
        href={href}
        isLoading={isLoading}
        className={className}
        id={id}
        {...otherProps}>
        {children}
        {isLoading && !disabled && renderLoading(actionType)}
      </StyledAnchorButton>
    )
  }

  if (to != null && to !== '') {
    return (
      <StyledRouterButton
        actionType={actionType}
        size={size}
        type={type}
        fullWidth={fullWidth}
        disabled={disabled}
        round={round}
        squared={squared}
        curved={curved}
        elevated={elevated}
        weight={weight}
        isLoading={isLoading}
        to={to}
        className={className}
        id={id}
        {...otherProps}>
        {children}
        {isLoading && !disabled && renderLoading(actionType)}
      </StyledRouterButton>
    )
  }

  return (
    <StyledButton
      actionType={actionType}
      size={size}
      type={type}
      isFlat={isFlat}
      fullWidth={fullWidth}
      disabled={disabled}
      round={round}
      squared={squared}
      curved={curved}
      elevated={elevated}
      weight={weight}
      isLoading={isLoading}
      onClick={handleOnClick}
      className={className}
      id={id}
      {...otherProps}>
      {children}
      {showCheckmark && <ButtonCheckmarkCircle />}
      {isLoading && !disabled && renderLoading(actionType)}
    </StyledButton>
  )
}
