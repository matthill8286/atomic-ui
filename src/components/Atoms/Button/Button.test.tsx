import React from 'react'
import { BrowserRouter as Router } from 'react-router-dom'
import { mountWithTheme, renderWithTheme } from '@/testRenderer'
import { Button } from './Button'

describe('Button', () => {
  const onClick = jest.fn()
  const buttonContent = 'Button '

  it('Should render a button', () => {
    const tree = renderWithTheme(<Button onClick={onClick}>{buttonContent}</Button>)
    expect(tree).toMatchSnapshot()
  })

  it('Should be clickable', () => {
    const tree = mountWithTheme(<Button onClick={onClick}>{buttonContent}</Button>)
    tree.find('button').simulate('click')
    expect(onClick.mock.calls.length).toEqual(1)
  })

  it('Should render an anchor', () => {
    const tree = renderWithTheme(<Button href="#">{buttonContent}</Button>)
    expect(tree).toMatchSnapshot()
  })

  it('Should render a router Link', () => {
    const tree = renderWithTheme(
      <Router>
        <Button to="/">{buttonContent}</Button>
      </Router>
    )
    expect(tree).toMatchSnapshot()
  })

  it('Should render a loading state', () => {
    const tree = renderWithTheme(
      <Button onClick={onClick} isLoading>
        {buttonContent}
      </Button>
    )
    expect(tree).toMatchSnapshot()
  })

  it('Should render a completed checkmark', () => {
    const tree = renderWithTheme(
      <Button onClick={onClick} showCheckmark>
        {buttonContent}
      </Button>
    )
    expect(tree).toMatchSnapshot()
  })
})
