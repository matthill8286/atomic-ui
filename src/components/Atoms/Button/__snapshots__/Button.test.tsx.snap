// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Should render a button 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 4px;
  box-sizing: border-box;
  outline: none;
  cursor: pointer;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  -webkit-appearance: none;
  padding: 12px 24px;
  position: relative;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-family: "Oswald","Raleway",Arial,sans-serif;
  font-weight: normal;
  font-size: 16px;
  height: 48px;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  box-shadow: none;
  background: #00C8DC;
  border: 2px solid #00C8DC;
  color: #FFFFFF;
}

.c0 .Icon-y4lcwx-0 {
  padding: 0 8px;
}

.c0 .Status-sc-1moqiqm-0 {
  position: absolute;
  top: -6px;
  right: -6px;
}

.c0:hover {
  box-shadow: 0 8px 10px 1px rgba(0,0,0,0.07),0 3px 14px 2px rgba(0,0,0,0.06),0 5px 5px -3px rgba(0,0,0,0.10);
}

.c0:hover {
  background: #00b5c8;
  border: 2px solid #00b5c8;
}

.c0 > span {
  color: inherit;
  display: block;
  text-align: center;
  width: 100%;
}

<button
  className="c0"
  disabled={false}
  onClick={[Function]}
  size="md"
  type="button"
>
  Button 
</button>
`;

exports[`Button Should render a completed checkmark 1`] = `
.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c2 svg {
  width: 16px;
  height: 16px;
  -webkit-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
  fill: currentColor;
}

.c2 svg g,
.c2 svg p {
  fill: currentColor;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 4px;
  box-sizing: border-box;
  outline: none;
  cursor: pointer;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  -webkit-appearance: none;
  padding: 12px 24px;
  position: relative;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-family: "Oswald","Raleway",Arial,sans-serif;
  font-weight: normal;
  font-size: 16px;
  height: 48px;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  box-shadow: none;
  background: #00C8DC;
  border: 2px solid #00C8DC;
  color: #FFFFFF;
}

.c0 .c1 {
  padding: 0 8px;
}

.c0 .Status-sc-1moqiqm-0 {
  position: absolute;
  top: -6px;
  right: -6px;
}

.c0:hover {
  box-shadow: 0 8px 10px 1px rgba(0,0,0,0.07),0 3px 14px 2px rgba(0,0,0,0.06),0 5px 5px -3px rgba(0,0,0,0.10);
}

.c0:hover {
  background: #00b5c8;
  border: 2px solid #00b5c8;
}

.c0 > span {
  color: inherit;
  display: block;
  text-align: center;
  width: 100%;
}

<button
  className="c0"
  disabled={false}
  onClick={[Function]}
  size="md"
  type="button"
>
  Button 
  <div
    className="c1 c2"
    height={16}
    width={16}
  >
    <svg
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12 2a10 10 0 1010 10A10 10 0 0012 2zm4.35 7.73l-5.62 5.62a.48.48 0 01-.7 0L7.65 13a.5.5 0 010-.71l.53-.53a.48.48 0 01.7 0l1.5 1.49 4.74-4.74a.5.5 0 01.7 0l.53.53a.5.5 0 010 .69z"
      />
    </svg>
  </div>
</button>
`;

exports[`Button Should render a loading state 1`] = `
.c1 {
  width: 20px;
  height: 20px;
  display: block;
  position: absolute;
}

.c1 div {
  box-sizing: border-box;
  display: block;
  position: absolute;
  width: 100%;
  height: 100%;
  border: 2px solid;
  border-radius: 50%;
  -webkit-animation: fEWCgj 1.2s cubic-bezier(0.5,0,0.5,1) infinite;
  animation: fEWCgj 1.2s cubic-bezier(0.5,0,0.5,1) infinite;
  border-color: #FFFFFF transparent transparent transparent;
}

.c1 div:nth-child(1) {
  -webkit-animation-delay: -0.45s;
  animation-delay: -0.45s;
}

.c1 div:nth-child(2) {
  -webkit-animation-delay: -0.3s;
  animation-delay: -0.3s;
}

.c1 div:nth-child(3) {
  -webkit-animation-delay: -0.15s;
  animation-delay: -0.15s;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 4px;
  box-sizing: border-box;
  outline: none;
  cursor: pointer;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  -webkit-appearance: none;
  padding: 12px 24px;
  position: relative;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-family: "Oswald","Raleway",Arial,sans-serif;
  font-weight: normal;
  font-size: 16px;
  height: 48px;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  box-shadow: none;
  background: #00C8DC;
  border: 2px solid #00C8DC;
  color: transparent;
}

.c0 .Icon-y4lcwx-0 {
  padding: 0 8px;
}

.c0 .Status-sc-1moqiqm-0 {
  position: absolute;
  top: -6px;
  right: -6px;
}

.c0:hover {
  box-shadow: 0 8px 10px 1px rgba(0,0,0,0.07),0 3px 14px 2px rgba(0,0,0,0.06),0 5px 5px -3px rgba(0,0,0,0.10);
}

.c0:hover {
  background: #00b5c8;
  border: 2px solid #00b5c8;
}

.c0 > span {
  color: inherit;
  display: block;
  text-align: center;
  width: 100%;
}

.c2 {
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  margin: auto;
}

<button
  className="c0"
  disabled={false}
  onClick={[Function]}
  size="md"
  type="button"
>
  Button 
  <div
    className="c1 c2"
    color="white"
    size={20}
  >
    <div />
    <div />
    <div />
    <div />
  </div>
</button>
`;

exports[`Button Should render a router Link 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 4px;
  box-sizing: border-box;
  outline: none;
  cursor: pointer;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  -webkit-appearance: none;
  padding: 12px 24px;
  position: relative;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-family: "Oswald","Raleway",Arial,sans-serif;
  font-weight: normal;
  font-size: 16px;
  height: 48px;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  box-shadow: none;
  background: #00C8DC;
  border: 2px solid #00C8DC;
  color: #FFFFFF;
}

.c0 .Icon-y4lcwx-0 {
  padding: 0 8px;
}

.c0 .Status-sc-1moqiqm-0 {
  position: absolute;
  top: -6px;
  right: -6px;
}

.c0:hover {
  box-shadow: 0 8px 10px 1px rgba(0,0,0,0.07),0 3px 14px 2px rgba(0,0,0,0.06),0 5px 5px -3px rgba(0,0,0,0.10);
}

.c0:hover {
  background: #00b5c8;
  border: 2px solid #00b5c8;
}

<a
  className="c0"
  href="/"
  onClick={[Function]}
>
  Button 
</a>
`;

exports[`Button Should render an anchor 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 4px;
  box-sizing: border-box;
  outline: none;
  cursor: pointer;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  -webkit-appearance: none;
  padding: 12px 24px;
  position: relative;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-family: "Oswald","Raleway",Arial,sans-serif;
  font-weight: normal;
  font-size: 16px;
  height: 48px;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  box-shadow: none;
  background: #00C8DC;
  border: 2px solid #00C8DC;
  color: #FFFFFF;
  -webkit-appearance: none !important;
}

.c0 .Icon-y4lcwx-0 {
  padding: 0 8px;
}

.c0 .Status-sc-1moqiqm-0 {
  position: absolute;
  top: -6px;
  right: -6px;
}

.c0:hover {
  box-shadow: 0 8px 10px 1px rgba(0,0,0,0.07),0 3px 14px 2px rgba(0,0,0,0.06),0 5px 5px -3px rgba(0,0,0,0.10);
}

.c0:hover {
  background: #00b5c8;
  border: 2px solid #00b5c8;
}

.c0:visited,
.c0:hover,
.c0:active {
  box-shadow: none;
}

<a
  className="c0"
  disabled={false}
  href="#"
  size="md"
  type="button"
>
  Button 
</a>
`;
